### PR TITLE
Separate focus management by build owner

### PR DESCRIPTION
**Pull request by krisgiesing**
Currently the focus manager is held by the singleton WidgetBinding.
This change places a focus manager in each build owner object,
which allows clients to run offscreen pipelines without disturbing
focus management for the main UI.

### CODE REVIEW

**Comment by krisgiesing** [comment]
@Hixie @cbracken @abarth @chinmaygarde

File: packages/flutter/test/widgets/independent_widget_layout_test.dart [hunk 1]
```diff
@@ -96,6 +96,51 @@ class TriggerableState extends State<TriggerableWidget> {
...
+    if (!_didAutofocus && widget.autofocus) {
+      _didAutofocus = true;
+      FocusScope.of(context).autofocus(widget.focusNode);
+    }
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+  }
```
Comment (abarth) [inline packages/flutter/test/widgets/independent_widget_layout_test.dart]: Remove?  This function just forwards to its base class.

**Comment by abarth** [comment]
LGTM

**Comment by abarth** [comment]
Thanks!

File: packages/flutter/test/widgets/independent_widget_layout_test.dart (hunk 1 above)
Comment (krisgiesing) [reply packages/flutter/test/widgets/independent_widget_layout_test.dart]: Yes, thanks, I should clean this up a bit (I cloned the similar class from test_focus and removed some code, but I can remove more).

**Pull request merged**